aten/src/ATen/SparseTensorImpl.cpp

```diff
@@ -79,7 +79,9 @@ int64_t SparseTensorImpl::storage_offset() const {
   AT_ERROR("sparse tensors do not have storage");
 }
 void SparseTensorImpl::set_indices_and_values_unsafe(const Tensor& indices, const Tensor& values) {
-  AT_CHECK(allow_tensor_metadata_change(), "set_indices_and_values_unsafe is not allowed on Tensor created from .data or .detach()");
+  AT_CHECK(allow_tensor_metadata_change(), "set_indices_and_values_unsafe is not allowed on Tensor created from .data or .detach(). "
+    "If you are using Python API, please make sure the Python function is called on the original tensor (not the tensor created from .data or .detach()), "
+    "and the function call is wrapped in a `with torch.no_grad():` block");
```

Comment:
I think it would be better if we could declare the instructions once, instead of copy pasting it this many places, in case we want to edit in the future